nautobot_ssot/integrations/servicenow/diffsync/models.py

```diff
@@ -82,7 +83,29 @@ def update(self, attrs):
         super().update(attrs)
         return self
 
-    # TODO delete() method
+    def delete(self):
+        """Delete an existing instance in ServiceNow if it does not exist in Nautobot. This code adds the ServiceNow object to the objects_to_delete[object_class] list. The actual delete occurs in the post-run method of adapter_servicenow.py."""
+        entry = self.diffsync.mapping_data[self.get_type()]
+        sn_resource = self.diffsync.client.resource(api_path=f"/table/{entry['table']}")
+        query = self.map_data_to_sn_record(data=self.get_identifiers(), mapping_entry=entry)
+        try:
+            sn_resource.get(query=query).one()
+        except pysnow.exceptions.MultipleResults:
+            self.diffsync.job.logger.error(
+                f"Unsure which record to update, as query {query} matched more than one item "
+                f"in table {entry['table']}"
+            )
+            return None
+        try:
+            self.diffsync.job.logger.warning(f"{self._modelname} {self.get_identifiers()} will be deleted.")
+            _object = sn_resource.get(query=query)
+            self.diffsync.objects_to_delete[self._modelname].append(_object)
+            print(self.diffsync.objects_to_delete[self._modelname])
+            super().delete()
+        except Exception as exception:
```

Comment:
Can you narrow this exception down a little so its not a blanket `except Exception`?

Reply:
After updating the code in the latest commit I don't believe a try/except clause is necessary here as I'm not having any issues with errors in my tests. I have eliminated this try/except clause in the latest commits.